pandas/tests/io/test_pytables.py

```diff
@@ -734,6 +734,59 @@ def test_put_compression_blosc(self):
             store.put('c', df, format='table', complib='blosc')
             tm.assert_frame_equal(store['c'], df)
 
+    def test_complibs_default_settings(self):
+        # GH15943
+        df = tm.makeDataFrame()
+
+        # Set complevel and check if complib is automatically set to
+        # default value
+        with ensure_clean_path(self.path) as tmpfile:
+            df.to_hdf(tmpfile, 'df', complevel=9)
+            result = pd.read_hdf(tmpfile, 'df')
+            tm.assert_frame_equal(result, df)
+
+            with tables.open_file(tmpfile, mode='r') as h5file:
+                for node in h5file.walk_nodes(where='/df', classname='Leaf'):
+                    assert node.filters.complevel == 9
+                    assert node.filters.complib == 'zlib'
+
+        # Set complib and check to see if compression is disabled
+        with ensure_clean_path(self.path) as tmpfile:
```

Comment:
I added the 3 cases where at least one of the parameters complib and complevel is missing. Setting complib and leaving complevel unset is one of those. I think these are good tests to see if the defaults behave as we discussed. 

I did not add tests to see what would happen if i set the parameters to illegal values, fx what happens if complevel is negative? 

I also added the testcase where i override the file-wide defaults.